plugins/epochstorage/plugin.go

```diff
@@ -296,99 +263,54 @@ func GetEpochVotersWeight(ei epoch.Index) (weights map[epoch.ECR]map[identity.ID
 	return weights
 }
 
-func getEpochContentStorage(ei epoch.Index) (*epochContentStorages, error) {
-	if ei < minEpochIndex {
-		return nil, errors.New("Epoch storage no longer exists")
-	}
-
-	epochContentsMutex.RLock()
-	stores, ok := epochContents[ei]
-	epochContentsMutex.RUnlock()
-
-	if !ok {
-		stores = newEpochContentStorage()
-		epochContentsMutex.Lock()
-		epochContents[ei] = stores
-		epochContentsMutex.Unlock()
-	}
-
-	return stores, nil
-}
-
-func newEpochContentStorage() *epochContentStorages {
-	db, _ := database.NewMemDB()
-
-	// keep data in temporary storage
-	return &epochContentStorages{
-		spentOutputs:   db.NewStore(),
-		createdOutputs: db.NewStore(),
-		transactionIDs: db.NewStore(),
-		blockIDs:       db.NewStore(),
-	}
-}
-
 func insertblockToEpoch(ei epoch.Index, blkID tangle.BlockID) error {
-	epochContentStorage, err := getEpochContentStorage(ei)
-	if err != nil {
-		return err
-	}
+	prefix := append([]byte{database.PrefixEpochsStorage, prefixBlockIDs}, ei.Bytes()...)
 
-	if err := epochContentStorage.blockIDs.Set(blkID.Bytes(), blkID.Bytes()); err != nil {
+	if err := baseStore.Set(append(prefix, blkID.Bytes()...), blkID.Bytes()); err != nil {
 		return errors.New("Fail to insert block to epoch store")
 	}
 	return nil
 }
 
 func removeblockFromEpoch(ei epoch.Index, blkID tangle.BlockID) error {
-	epochContentStorage, err := getEpochContentStorage(ei)
-	if err != nil {
-		return err
-	}
+	prefix := append([]byte{database.PrefixEpochsStorage, prefixBlockIDs}, ei.Bytes()...)
 
-	if err := epochContentStorage.blockIDs.Delete(blkID.Bytes()); err != nil {
+	if err := baseStore.Delete(append(prefix, blkID.Bytes()...)); err != nil {
 		return errors.New("Fail to remove block from epoch store")
 	}
 	return nil
 }
 
 func insertTransactionToEpoch(ei epoch.Index, txID utxo.TransactionID) error {
-	epochContentStorage, err := getEpochContentStorage(ei)
-	if err != nil {
-		return err
-	}
+	prefix := append([]byte{database.PrefixEpochsStorage, prefixTransactionIDs}, ei.Bytes()...)
 
-	if err := epochContentStorage.transactionIDs.Set(txID.Bytes(), txID.Bytes()); err != nil {
+	if err := baseStore.Set(append(prefix, txID.Bytes()...), txID.Bytes()); err != nil {
```

Comment:
I wouldn't append the prefix so directly. You can use `baseStore.WithRealm` to obtain a copy of the same underlying KVStore with the prefix set for you; so each object you operate on will always have that realm (prefix).